pkg/proxy/probe/probe.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2021 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	s "github.com/cortexlabs/cortex/pkg/lib/strings"
+	"github.com/cortexlabs/cortex/pkg/proxy"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	kcore "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	_defaultInitialDelaySeconds = 1
```

Comment:
Is this currently being used?

Reply:
I just realized it is not, but maybe it does not make sense to use it (?)

Reply:
I can add it, it shouldn't be hard, but I need to add a mutex to check if it is the first time the probe is executed